layer1/P.cpp

```diff
@@ -615,9 +615,9 @@ static T const* get_member_pointer(S const* instance, size_t offset)
  * Should be equivalent to:
  *   OBAtom::GetExplicitValence() [Open Babel 3.0]
  */
-static int getExplicitValence(ObjectMolecule const* obj, size_t atm)
+float getExplicitValence(ObjectMolecule const* obj, size_t atm)
```

Comment:
I think the type here should be left as int (or unsigned int); would be consistent with the corresponding functions from OpenBabel and RDKit which this function is based off of. Can you instead just cast either the numerator or denominator to float if you need a floating result below?

Reply:
The delocalization is indicated by the float number because otherwise it were a int, it would cast implicitly because the "+3/2" rule.

Reply:
I haven't looked into this too much, but I thought the decision to make it `int` was intentional even with this rule. What makes PyMOL's case different that we have to return `float` here while the two APIs this is similar to (from `OpenBabel` and `RDKit`) still return `int`?